src/core/qfieldxmlhttprequest.cpp

```diff
@@ -0,0 +1,794 @@
+/***************************************************************************
+qfieldxmlhttprequest.cpp - QFIELDXMLHTTPREQUEST
+
+---------------------
+begin                : 15.12.2025
+copyright            : (C) 2025 by Kaustuv Pokharel
+email                : kaustuv (at) opengis.ch
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************/
+
+#include "qfieldxmlhttprequest.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QHttpMultiPart>
+#include <QHttpPart>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonParseError>
+#include <QJsonValue>
+#include <QNetworkReply>
+#include <QTimer>
+#include <qfieldcloudutils.h>
+
+#include <algorithm>
+#include <fileutils.h>
+#include <networkmanager.h>
+#include <networkreply.h>
+#include <urlutils.h>
+
+QFieldXmlHttpRequest::QFieldXmlHttpRequest( QObject *parent )
+  : QObject( parent )
+{
+}
+
+QFieldXmlHttpRequest::~QFieldXmlHttpRequest()
+{
+  // Destructor must be silent (no QML callbacks / no signals)
+  cleanupReply( /*abortNetwork=*/true );
+}
+
+QFieldXmlHttpRequest *QFieldXmlHttpRequest::newRequest( QObject *parent )
+{
+  return new QFieldXmlHttpRequest( parent );
+}
+
+void QFieldXmlHttpRequest::setTimeout( int ms )
+{
+  const int v = ms > 0 ? ms : 0;
+  if ( mTimeoutMs == v )
+    return;
+
+  mTimeoutMs = v;
+  emit timeoutChanged();
+}
+
+void QFieldXmlHttpRequest::setAutoDelete( bool v )
```

Comment:
What does v stand for? We want our code to be easy to read for newcomers, let's spell this out loud.